### ISSUE TITLE
[JENKINS-29243] Allow for specifying the '--exec-path' parameter

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/git-client-plugin#1482]
<p>When a Git installation is transferred from one host to another, it has the <tt>git_exec_path</tt> parameter hardened as specified when running <tt>make</tt>. In other words, the installation is not fully portable unless it is compiled on the same machine or the install directory is kept after installing. Therefore, commands that are based on scripts, such as <tt>git submodule</tt> do not work.</p>

<p>As a workaround one can set the <tt>--exec-path</tt> parameter. This would make the automatic installation of git using "extract a &#42;.zip/&#42;.tar.gz archive" useful and allow different versions of git run in parallel on the same machine without having the need to install it on OS level.</p>

<p><tt>git-client-plugin</tt> should allow for specifying this parameter as a directory relative to the binary archive extraction root in the Git settings. It should then be passed to any git command as the full absolute path.</p>

<p>To illustrate, here's how it could be configured then:</p>

<table><tr><td><b>Install automatically</b></td></tr><tr><td><div class='table-wrap'>
<table class='confluenceTable'><tbody>
<tr>
<td class='confluenceTd'>Label</td>
<td class='confluenceTd'> </td>
</tr>
<tr>
<td class='confluenceTd'>Download URL</td>
<td class='confluenceTd'><tt><a href="http://some.url.to/git-bin-version.tar.gz" class="external-link" target="_blank" rel="nofollow noopener">http://some.url.to/git-bin-version.tar.gz</a></tt></td>
</tr>
<tr>
<td class='confluenceTd'>Subdirectory </td>
<td class='confluenceTd'><tt>bin/git</tt></td>
</tr>
<tr>
<td class='confluenceTd'>Exec-Path</td>
<td class='confluenceTd'><tt>libexec/git-core</tt></td>
</tr>
</tbody></table>
</div></td></tr></table>

<p>If left empty, the parameter should rather not be passed at all.</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/xasx.png" title="xasx's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=xasx">xasx</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-29243" target="_blank">Allow for specifying the '--exec-path' parameter</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>component(s)</b>: git-client-plugin
<li><b>resolution</b>: Won't Do
<li><b>resolved</b>: 2025-04-29T17:35:25+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 2
<li><b>imported</b>: 20251211-071809
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>When a Git installation is transferred from one host to another, it has the <tt>git_exec_path</tt> parameter hardened as specified when running <tt>make</tt>. In other words, the installation is not fully portable unless it is compiled on the same machine or the install directory is kept after installing. Therefore, commands that are based on scripts, such as <tt>git submodule</tt> do not work.</p>

<p>As a workaround one can set the <tt>--exec-path</tt> parameter. This would make the automatic installation of git using "extract a &#42;.zip/&#42;.tar.gz archive" useful and allow different versions of git run in parallel on the same machine without having the need to install it on OS level.</p>

<p><tt>git-client-plugin</tt> should allow for specifying this parameter as a directory relative to the binary archive extraction root in the Git settings. It should then be passed to any git command as the full absolute path.</p>

<p>To illustrate, here's how it could be configured then:</p>
<div class="panel" style="border-width: 1px;"><div class="panelHeader" style="border-bottom-width: 1px;"><b>Install automatically</b></div><div class="panelContent">
<div class='table-wrap'>
<table class='confluenceTable'><tbody>
<tr>
<td class='confluenceTd'>Label</td>
<td class='confluenceTd'>&nbsp;</td>
</tr>
<tr>
<td class='confluenceTd'>Download URL</td>
<td class='confluenceTd'><tt><a href="http://some.url.to/git-bin-version.tar.gz" class="external-link" target="_blank" rel="nofollow noopener">http://some.url.to/git-bin-version.tar.gz</a></tt></td>
</tr>
<tr>
<td class='confluenceTd'>Subdirectory </td>
<td class='confluenceTd'><tt>bin/git</tt></td>
</tr>
<tr>
<td class='confluenceTd'>Exec-Path</td>
<td class='confluenceTd'><tt>libexec/git-core</tt></td>
</tr>
</tbody></table>
</div>
</div></div>

<p>If left empty, the parameter should rather not be passed at all.</p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-29243?focusedId=320078&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-320078">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>If this is ever implemented, refer to the <a href="https://git-scm.com/book/gr/v2/Git-Internals-Environment-Variables" class="external-link" target="_blank" rel="nofollow noopener">git internal variables page</a> and the <a href="https://stackoverflow.com/questions/7650494/setting-up-a-custom-git-installation" class="external-link" target="_blank" rel="nofollow noopener">stackoverflow article</a>.</p>
</pre>
</details>

</ul>
</details>

<p>If this is ever implemented, refer to the <a href="https://git-scm.com/book/gr/v2/Git-Internals-Environment-Variables" class="external-link" target="_blank" rel="nofollow noopener">git internal variables page</a> and the <a href="https://stackoverflow.com/questions/7650494/setting-up-a-custom-git-installation" class="external-link" target="_blank" rel="nofollow noopener">stackoverflow article</a>.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-29243?focusedId=454217&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-454217">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>No plan to implement this.</p>
</pre>
</details>

</ul>
</details>

<p>No plan to implement this.</p>
